cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+        List<String> addedPartitionList = new ArrayList<>(sourcePartitionList);
+        addedPartitionList.removeAll(previousPartitionList);
+        List<String> removedBusinessPartitionList = new ArrayList<>();
+        List<String> addedBusinessPartitionList = new ArrayList<>();
+
+        for (String addedPartition : addedPartitionList) {
+            addedBusinessPartitionList.add(addedPartition.substring(0, addedPartition.lastIndexOf("/")));
+        }
+
+        for (String deletedPartition : deletedPartitionList) {
+            removedBusinessPartitionList.add(deletedPartition.substring(0, deletedPartition.lastIndexOf("/")));
+        }
+        Set<String> businessPartitionSet = new HashSet<>();
+        businessPartitionSet.addAll(removedBusinessPartitionList);
+        businessPartitionSet.addAll(addedBusinessPartitionList);
+
+        for (String editedBusinessPartition : businessPartitionSet) {
+            LOGGER.debug("editedBusinessPartition = " + editedBusinessPartition);
+        }
+        return businessPartitionSet;
+    }
+
+    private List<String> getCurrentSourceColumnList(String sourceDb, String table, String sourceHiveConnection) throws Exception {
+        List<String> sourceColumnList = new ArrayList<>();
+        DatabaseMetaData metaData = getHiveJDBCConnection(sourceDb,sourceHiveConnection).getMetaData();
+        ResultSet rsColumns = metaData.getColumns(null, sourceDb, table, null);
+        while (rsColumns.next()) {
+            String columnName = rsColumns.getString("COLUMN_NAME");
+            String dataType = rsColumns.getString("TYPE_NAME");
+            sourceColumnList.add(columnName + " " + dataType);
+        }
+        rsColumns.close();
+        return sourceColumnList;
+    }
+
+    private List<String> getDestColumnList(String processId, String table) {
+        //obtaining the column data corresponding to the previous execution for comparison with current partition data
+        ProcessLog processLog = new ProcessLog();
+        List<ProcessLogInfo> columnLogInfos = processLog.listLastInstanceRef(Integer.parseInt(processId), table + " column");
+        List<String> previousColumnList = new ArrayList<>();
+        for (ProcessLogInfo processLogInfo : columnLogInfos) {
+            previousColumnList.add(processLogInfo.getMessage());
+        }
+        return previousColumnList;
+    }
+
+    private List<String> getNewRegularColumnsAtSourceList(List<String> sourceColumnList, List<String> previousColumnList) {
+        List<String> addedColumnList = new ArrayList<>(sourceColumnList);
+        //TODO: if previous list is empty it means there are no log entries in process log currently. But this could be the first run of the whole migration program, hence there will obviously be no log entries, in this case add logic to test destination columns using gethivejdbcconnection or you can skip this logic entirely and get columns always from hive metadata.
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1135)